htdocs/product/class/api_products.class.php

```diff
@@ -920,12 +921,12 @@ public function getAttributes($sortfield = "t.ref", $sortorder = 'ASC', $limit =
         $result = $this->db->query($sql);
 
         if (!$result) {
-            throw new RestException(503, 'Error when retrieve product list : '.$db->lasterror());
+            throw new RestException(503, 'Error when retrieve product attribute list : '.$db->lasterror());
         }
 
         $return = [];
         while ($result = $this->db->fetch_object($query)) {
-            $tmp = new stdClass();
+            $tmp = new ProductAttribute($this->db);
```

Comment:
Using ProductAttribute class is counterproductive. I used stdClass to have clean result. If you use ProductAttribute class, you have to clean the object at each loop.